### DISCUSSION TITLE
How to add serverpod to an existing flutter project?

### DISCUSSION THREAD

**Discussion by EMUNES** [serverpod/serverpod#2929] [Q&A] [answered]
Hello, all. I have a flutter project, and I'm looking for a proper backend. I have checked the documentation, and I find no information about adding serverpod in my current project. I have changed some files in Android config files, so creating a new one could be error-prone. Any clue how to properly add serverpod to an existing project. Thanks in advance.

**Comment by adiltirur** [comment, marked as answer]
@EMUNES, adding Serverpod to an existing Flutter project is quite straightforward. Follow the steps:

Create a new Serverpod project: Run the command:

`serverpod create`

This will generate three components:

A Flutter client project
A Serverpod backend project
A Flutter app project

Keep only the backend project: Since you already have a Flutter app, you can delete the generated Flutter app and client project folders. You’ll only need the backend project folder.

Configure the client generation path: Open the generator.yaml file in the Serverpod backend project and update the client_package_path to point to a folder within your existing Flutter app, such as your-flutter-app/packages/client. Here’s an example:


`client_package_path: ../../your-flutter-app/packages/your_client_name`

Replace your_client_name with the name that was already there.

Generate the client: Run the following command in the backend project directory to generate the client code:

`serverpod generate`

This will create the client package in the specified location within your Flutter project.

Add the client to your Flutter project: Finally, add a dependency for the generated client package in your pubspec.yaml file:

```
dependencies:
  your_client_name:
    path: ./packages/your_client_name
```

Run flutter pub get to install the client package.

by following the above steps, you should be able to add Serverpod to existing flutter project

**Reply by EMUNES** [reply]
Thanks for this detailed instructions! It is really helpful. I will try those steps:)

**Reply by Pak-Man926** [reply]
This is a little misleading but maybe worked before. I tried but it did not work for me, instead I just run serverpod create (create a file name similar to your flutter project) then deleted the flutter files and replaced them with my already ready server files. Check the pubsec.yaml of your flutter file and add these two dependencies,
serverpod_flutter: 2.6.0 or the latest version
your_project_client:
                   path:  ../your_projectclient_folder

Just that then serverpod generate to confirm everything works and you'll be ready to continue development with serverpod.

**Comment by EMUNES** [comment]
If I delete the client folder, and point serverpod server to another client path, it will report error:

```
An error occurred while parsing the server config file: Failed to load client pubspec.yaml. If you are using a none
default path it has to be specified in the config/generator.yaml file!
```

The client path is an existing empty folder. And it will report no pubspec.yaml found. How to solve this issue, maybe we need to move the generated client package instead of deleting it?

**Reply by PGScully** [reply]
Do not delete the client folder.  This is a library that handles all the communication with the server.  It should be included in the Flutter app's `pubspec.yaml`:

```
dependencies:
  foo_client:
    path: ../foo_client
```

(Update name and path as needed.)  You'll need to import and create the client in your Flutter app - see the generated Flutter app for an example of how to do this.

Every time you update the server (e.g. changing tables or endpoints, etc) run `serverpod generate` and it will rebuild the client with your changes.

**Reply by adiltirur** [reply]
@EMUNES sorry about the oversight, as mentioned by @PGScully just copy this folder to your desired location and then everytime you run `serverpod generate`, this should create the required generated codes. But please make sure if follow the steps that I mentioned for example to include `client_package_path` in the generator.yaml,

I hope you got it working

**Reply by EMUNES** [reply]
Yes, moving the client directory is working. Thanks for those insights.

**Comment by cstrail** [comment]
I have the backend mini project running on my remote server.  I have a local flutter app I am wanting to add serverpod to.  I cannot find the generator.yaml file to modify.
